srv/salt/_modules/packagemanager.py

```diff
@@ -217,6 +218,32 @@ def _updates_needed(self):
         log.info('No Update Needed')
         return False
 
+    def _parse_xml(self, xml, find_str='.//update'):
+        """
+        utils method to parse xml from a str
+        """
+        root = ET.fromstring(xml)
+        ret = list()
+        for child in root.findall(find_str):
+            ret.append(child.attrib)
+        return ret
+
+    # pylint: disable=dangerous-default-value
+    def list_updates(self, _filter=[]):
+        """
+        List all pending updates (transformed from XML)
+        """
+        self._refresh()
```

Comment:
`zypper ref` here on my vagrant test setup (with all local repos for SLE and SES Pool and Updates) takes 1.171 seconds. On my desktop system though, with several remote repos (opensuse.org, etc.), `zypper ref` takes 16.4 seconds, which would make this rather more unpleasant. I'm not sure there's really any way around this though (the repos need to be refreshed to know if there's actually updates; if we skipped the explicit refresh, it would only work properly if the repos were configured to auto-refresh, and they might not be).

Reply:
I tested this on my vagrant setup with 'local' (in-house but non-host-local) repos and the whole process including the refresh took 3 seconds.

I notices that zypper returns something like:

`Repository 'repo_name' is out-of-date. You can run 'zypper refresh' as root to update it.`

That's something we could look for, which would in turn require to run zypper lu twice..
I think for people running their DCs with a SMT+local mirror it's gonna be fine, but people that have mirrors which are a bit further away it will be a considerable time waste..

I do think that it's worse to miss and update at the hence miss the warning and then actually apply the update in a later stage. `_updates_needed` also invokes `_refresh` which means that when we leave out the `_refresh` in `list_updates`, we actually have a change to miss to emit a warning

Reply:
I was wondering the same. Though I'm not a fan of preventing a stage for this...seems confusing. Ideally we would print the validation messages again after everything run to make it visible, but that seems like a lot of work.

Reply:
Mhh, I don't know if we have many other options.. Either we hazard the consequences of having a disconnect between what we show in the validation and actually do in the stage or spend more time on (unnecessary)refreshing of repos.. Tough call..